### ISSUE TITLE
404s for built .js files in development environment

### ISSUE THREAD

**Issue by bradurani** [rwz/ember-cli-rails#42]
I've set this up using the simplest possible configuration exactly like the readme suggests, but I'm getting 404s for:
/javascripts/app_name/app-name.js
/javascripts/app_name/vendor.js

When I run rake assets:precompile I see what looks like the build files under public/assets/app_name/vendor-xxxxxxxxxxxxxxxxxx.js etc., so it looks like that works (I have no way to test).

Can you give me a better idea of what I should be seeing in development mode? Should the ember build output be displaying anywhere? Where should I be looking for temporary files? Are their certain asset pipeline config settings that break this?

**Comment by rwz** [comment]
What files do you have in `dist` after you manually run `ember build`?

**Comment by bradurani** [comment]
app-name.js
vendor.js
and a number test-related and css files I don't care about

**Comment by bradurani** [comment]
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/advisor-document-center.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/advisor-document-center.js
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/application.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/dashboard.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/failed.png
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/feedback.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/passed.png
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/test-loader.js
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/test-support.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/test-support.js
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/upside.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/vendor.css
/Users/bradurani/src/upside/app/advisor_document_center/dist/assets/vendor.js

**Comment by bradurani** [comment]
Figured it out by looking at your code (very nice BTW). I had config.consider_all_requests_local = false

**Comment by rwz** [comment]
Cool. Why did you have `consider_all_requests_local=false` in the first place? Maybe there is a better way for us to detect non-production environment?

**Comment by bradurani** [comment]
I think I was testing error handling code I wrote and forgot to turn it back on. It's probably an unsafe assumption to assume that everyone has that set to true. I opened a PR to reflect this in the README.md so at a minimum it's documented behavior. Maybe the best way to handle this is to not check `consider_all_requests_local` but rather let users configure which environments use `--watch` in the initializer? Or does using `--watch` when `consider_all_requests_local=false` cause issues?
